service/controller/v19/adapter/guest_subnets.go

```diff
@@ -1,63 +1,44 @@
 package adapter
 
 import (
-	"net"
+	"fmt"
 
 	"github.com/giantswarm/aws-operator/service/controller/v19/key"
-	"github.com/giantswarm/ipam"
-	"github.com/giantswarm/microerror"
 )
 
+type Subnet struct {
+	Index               int
+	AvailabilityZone    string
+	CIDR                string
+	Name                string
+	MapPublicIPOnLaunch bool
+}
+
 type GuestSubnetsAdapter struct {
-	PublicSubnetAZ                   string
-	PublicSubnetCIDR                 string
-	PublicSubnetName                 string
-	PublicSubnetMapPublicIPOnLaunch  bool
-	PrivateSubnetAZ                  string
-	PrivateSubnetCIDR                string
-	PrivateSubnetName                string
-	PrivateSubnetMapPublicIPOnLaunch bool
+	PublicSubnets  []Subnet
+	PrivateSubnets []Subnet
 }
 
 func (s *GuestSubnetsAdapter) Adapt(cfg Config) error {
```

Comment:
I'd sort the AZs here instead in `key.AvailabilityZones`. It's kind of indirect to rely on that.